consumer/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/consumer/service/oppgavebehandling/rest/RestOppgaveBehandlingServiceImpl.kt

```diff
@@ -0,0 +1,415 @@
+                tilordnetRessurs = request.ansvarligIdent.coerceBlankToNull(),
+                tildeltEnhetsnr = request.ansvarligEnhetId.coerceBlankToNull(),
+                beskrivelse = request.beskrivelse.coerceBlankToNull(),
+                temagruppe = request.temagruppe.coerceBlankToNull(),
+                tema = request.tema.coerceBlankToNull(),
+                behandlingstema = behandling.map(Behandling::getBehandlingstema).orElse(null),
+                oppgavetype = oppgaveType,
+                behandlingstype = behandling.map(Behandling::getBehandlingstype).orElse(null),
+                aktivDato = LocalDate.now(),
+                fristFerdigstillelse = request.oppgaveFrist,
+                prioritet = PostOppgaveRequestJsonDTO.Prioritet.valueOf(stripTemakode(request.prioritet))
+            )
+        )
+
+        val oppgaveId = response.id ?: throw IllegalStateException("Response inneholdt ikke oppgaveId")
+        return OpprettOppgaveResponse(oppgaveId.toString())
+    }
+
+    override fun opprettSkjermetOppgave(request: OpprettSkjermetOppgaveRequest?): OpprettOppgaveResponse {
+        requireNotNull(request)
+        val behandling = kodeverksmapperService.mapUnderkategori(request.underkategoriKode)
+        val oppgaveType = kodeverksmapperService.mapOppgavetype(request.oppgavetype)
+        val aktorId = getAktorId(request.fnr) ?: throw IllegalArgumentException("Fant ikke aktorId for ${request.fnr}")
+
+        val response = apiClient.opprettOppgave(
+            xminusCorrelationMinusID = correlationId(),
+            postOppgaveRequestJsonDTO = PostOppgaveRequestJsonDTO(
+                opprettetAvEnhetsnr = request.opprettetavenhetsnummer.coerceBlankToNull(),
+                aktoerId = aktorId,
+                behandlesAvApplikasjon = "FS22",
+                beskrivelse = request.beskrivelse.coerceBlankToNull(),
+                temagruppe = null,
+                tema = request.tema.coerceBlankToNull(),
+                behandlingstema = behandling.map(Behandling::getBehandlingstema).orElse(null),
+                oppgavetype = oppgaveType,
+                behandlingstype = behandling.map(Behandling::getBehandlingstype).orElse(null),
+                aktivDato = LocalDate.now(),
+                fristFerdigstillelse = request.oppgaveFrist,
+                prioritet = PostOppgaveRequestJsonDTO.Prioritet.valueOf(stripTemakode(request.prioritet))
+            )
+        )
+
+        val oppgaveId = response.id ?: throw IllegalStateException("Response inneholdt ikke oppgaveId")
+        return OpprettOppgaveResponse(oppgaveId.toString())
+    }
+
+    override fun hentOppgave(oppgaveId: String?): Oppgave {
+        requireNotNull(oppgaveId)
+        val oppgave = hentOppgaveJsonDTO(oppgaveId)
+
+        return mapTilOppgave(oppgave)
+    }
+
+    override fun tilordneOppgaveIGsak(oppgaveId: String?, temagruppe: Temagruppe?, saksbehandlersValgteEnhet: String?) {
```

Comment:
Burde ikke alle funksjoner som før hadde `IGsak` på slutten, få fjernet `IGsak` siden vi nå ikke skal gå via Gsak mer? F.eks. `tilordneOppgaveIGsak` --> `tilordneOppgave`

Reply:
Jupp, det burde de. :) 
Men i en overgangsfase ønsker vi å kunne bruke en feature-toggle for å bytte mellom soap og rest implementasjonen, og mens det pågår er det veldig kjekt at `OppgaveBehandlingServiceImpl` og `RestOppgaveBehandlingServiceImpl` implementerer de samme metodene. 

En del av jobben jeg gjorde var derfor å få `RestOppgaveBehandlingServiceImpl` til å implementere samme interface som `OppgaveBehandlingServiceImpl`, og på den måten trenger ikke resten av applikasjonen å forholde seg til hvorvidt soap eller rest versjonen er aktiv.